.github/workflows/release.yaml

```diff
@@ -0,0 +1,280 @@
+    needs:
+      - extract-version-details
+      - obtain-approval
+    permissions:
+      contents: write
+      id-token: write
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.extract-version-details.outputs.full-version }}
+      - name: Publish the full version release
+        run: |
+          gh release edit ${{ needs.extract-version-details.outputs.full-version }} \
+            --draft=false \
+            --latest=${{ inputs.kind == 'production' }}
+      - name: Push the aliased tags
+        if: ${{ needs.extract-version-details.outputs.aliased-version != '' }}
+        run: |
+          git tag --force ${{ needs.extract-version-details.outputs.aliased-version }}
+          git push --tags --force origin ${{ needs.extract-version-details.outputs.aliased-version }}
+      - id: edit-aliased-version-release
+        name: Update the aliased release if it already exists
+        if: ${{ needs.extract-version-details.outputs.aliased-version != '' }}
+        continue-on-error: true
+        run: gh release edit ${{ needs.extract-version-details.outputs.aliased-version }} --tag ${{ needs.extract-version-details.outputs.aliased-version }}
+      - name: Create the aliased release otherwise
+        if: ${{ steps.edit-aliased-version-release.outcome == 'failure' && needs.extract-version-details.outputs.aliased-version != ''}}
+        run: |
+          gh release create ${{ needs.extract-version-details.outputs.aliased-version }} \
+            --notes "The Captain ${{ needs.extract-version-details.outputs.aliased-version }} release and tag exist to provide an easy way to download the latest ${{ needs.extract-version-details.outputs.aliased-version }}.x.x release of Captain. For example, you can always download the latest Linux x86 ${{ needs.extract-version-details.outputs.aliased-version }} release at this URL: https://github.com/rwx-research/captain/releases/download/${{ needs.extract-version-details.outputs.aliased-version }}/captain-linux-x86_64. (Look at the assets attached to this release to find the other available downloads.) This release and its assets are updated whenever a new ${{ needs.extract-version-details.outputs.aliased-version }}.x.x version of Captain is released." \
+            --title "Captain ${{ needs.extract-version-details.outputs.aliased-version }}"
+      - name: Push the assets from the full version release to the aliased release
+        if: ${{ needs.extract-version-details.outputs.aliased-version != '' }}
+        run: |
+          gh release download ${{ needs.extract-version-details.outputs.full-version }}
+          gh release upload ${{ needs.extract-version-details.outputs.aliased-version }} captain-* --clobber
+
+  update-cloudfront:
+    runs-on: ubuntu-latest
+    environment: Production
+    if: ${{ always() && needs.extract-version-details.result == 'success' && needs.publish-release.result == 'success' }}
```

Comment:
💭 I don't understand this conditional. The always() feels at odds with the success requirements

Reply:
I can probably leave a comment